addons/web/static/tests/views/pivot_view.test.js

```diff
@@ -0,0 +1,4551 @@
+    await contains(".o-dropdown--menu .dropdown-item:eq(5)").click();
+
+    // Click on the two header dropdown togglers
+    await contains("thead th.o_pivot_header_cell_closed:eq(0)").click();
+    await contains("thead th.o_pivot_header_cell_closed:eq(1)").click();
+
+    expect(".o-dropdown--menu").toHaveCount(1);
+});
+
+test("Server order is kept by default", async () => {
+    expect.assertions(1);
+
+    let isSecondReadGroup = false;
+    onRpc("read_group", () => {
+        if (isSecondReadGroup) {
+            return Promise.resolve([
+                {
+                    customer: [2, "Second"],
+                    foo: 18,
+                    __count: 2,
+                    __domain: [["customer", "=", 2]],
+                },
+                {
+                    customer: [1, "First"],
+                    foo: 14,
+                    __count: 2,
+                    __domain: [["customer", "=", 1]],
+                },
+            ]);
+        }
+        isSecondReadGroup = true;
+    });
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        arch: `
+			<pivot>
+				<field name="customer" type="row"/>
+				<field name="foo" type="measure"/>
+			</pivot>`,
+    });
+
+    const values = [
+        "32", // Total Value
+        "18", // Second
+        "14", // First
+    ];
+    expect(getCurrentValues()).toBe(values.join());
+});
+
+test("pivot rendering with boolean field", async () => {
+    expect.assertions(4);
+
+    Partner._fields.bar = fields.Boolean({
+        string: "bar",
+        store: true,
+        searchable: true,
+        aggregator: "bool_or",
+    });
+    Partner._records = [
+        { id: 1, bar: true, date: "2019-12-14" },
+        { id: 2, bar: false, date: "2019-05-14" },
+    ];
+
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        arch: `
+			<pivot>
+				<field name="date" type="row" interval="day"/>
+				<field name="bar" type="col"/>
+				<field name="bar" string="SLA status Failed" type="measure"/>
+			</pivot>`,
+    });
+
+    expect('tbody tr:contains("2019-12-14")').toHaveCount(1);
+    expect('tbody tr:contains("2019-12-14") [type="checkbox"]').toBeChecked();
+    expect('tbody tr:contains("2019-05-14")').toHaveCount(1);
+    expect('tbody tr:contains("2019-05-14") [type="checkbox"]').not.toBeChecked();
+});
+
+test.tags("desktop")("empty pivot view with action helper", async () => {
+    Partner._views["pivot,false"] = `<pivot>
+		<field name="product_id" type="measure"/>
+		<field name="date" interval="month" type="col"/>
+	</pivot>`;
+    Partner._views["search,false"] = `<search>
+		<filter name="small_than_0" string="Small Than 0" domain="[('id', '&lt;', 0)]"/>
+	</search>`;
+
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        context: { search_default_small_than_0: true },
+        noContentHelp: markup(`<p class="abc">click to add a foo</p>`),
```

Comment:
The string in the function is a constant. What's more, the function is used in a unit test. The risk is zero.